src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -110,6 +113,22 @@ public static class DependencyGraphRestoreUtility
             return new List<RestoreSummary>();
         }
 
+        private static void PersistDGSpec(DependencyGraphSpec dgSpec)
+        {
+            var filePath = Path.Combine(
+                    NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp),
+                    "nuget-dg",
+                    "nugetSpec.dg");
+
+            // delete nugetSpec.dg if exists
+            FileUtility.Delete(filePath);
```

Comment:
do any of these throw?

Reply:
I'm not expecting but how many times things go as per our expectation :) so i better safeguard.

Reply:
besides, only place it could throw is when saving that dg spec file and if that happens, then we would have way more bigger problem.

Reply:
yeah...but FileUtility.Delete throws on the 3rd try. 

I'd wrap the whole thing in a try-catch block and ignore it if it fails.